services/search-builder.js

```diff
@@ -24,6 +24,17 @@ function SearchBuilder(model, opts, params) {
             !field.reference) {
             q[key] = new RegExp('.*' + params.search + '.*', 'i');
             orQuery.$or.push(q);
+          } else if (field && _.isArray(field.type) &&
+            !field.reference && parseInt(params.searchExtended)) {
+            var elemMatch = { $elemMatch: { $or: [], } };
+
+            field.type[0].fields.forEach(function(subField) {
```

Comment:
I think you need to apply the search on restricted type of field (string ??)
unless you want this kind of error:

![screen shot 2017-08-21 at 14 30 24](https://user-images.githubusercontent.com/664261/29519365-75edd3ea-867d-11e7-9134-1fc4ffcbfde2.png)